sync/downstream.py

```diff
@@ -159,30 +159,22 @@ def next_action(self):
         latest_try_push = self.latest_valid_try_push
 
         if not latest_try_push:
-            logger.debug("Sync for PR %s has no valid try push" % self.pr)
-            return DownstreamAction.try_push
+            if self.requires_stability_try:
+                logger.debug("Sync for PR %s requires a stability try push" % self.pr)
+                pr = env.gh_wpt.get_pull(self.pr)
+                if pr.merged:  # Wait till PR is merged to push stability run
```

Comment:
I think this condition should be outside the initial `if` i.e. if we're not merged always return `wait_approved` (which could be renamed `wait_upstream` or something now it's not about being approved).

Reply:
That sounds like a good idea, will also need to move the `manual_fix` and `wait_try` states into this new if.